index.js

```diff
@@ -8,32 +8,48 @@ var Reader = module.exports = function(options) {
   options = options || {}
   this.offset = 0
   this.lastChunk = false
-  this.chunk = null
+  this.chunk = Buffer.alloc(4);
```

Comment:
I realized I can just go back to assigning this to null and just keeping a handle to the first chunk added.